### ISSUE TITLE
DockerTasker should be standardized with Nomad (Ship)

### ISSUE THREAD

**Issue by meganokeefe** [sigma-dev/sigma#174]
### problem 

The way we do Progress/Outputs Shipping between the two taskers is different right now -- 

NomadTasker has a shipper script that it runs as an exec sidecar task for every job. (Ie. a curl script for progress/output is run alongside every plugin container). Nomad controls when this `curl` happens using its own scheduler.

But DockerTasker does these `curl`s on-demand-- specifically, manager calls `DockerTasker.Ship()` at the configurable service update interval, which gets progress/output into the DB, and then on the next-loop around, manager has those fields at its disposal. 

### solution 

Because we're locked into this pattern for Nomad, we should change DockerTasker to, instead of having `tasker.Ship`, to spawn a goroutine that monitors the filesystem's progress/output for changes, and `curl`s to the sigma API anytime there's a change. 

This goroutine should also (similar to nomad) gracefully shutdown when `ContainerStop` happens, `curl`ing the progress/outputs one last time. 

Basically remove the `Ship()` function from the tasker interface completely.
